### PR TITLE
Better error message when importing missing stack

### PR DESCRIPTION
**Pull request by adamconnelly**
## Description of the change

The terraform import handling for stacks was reusing the `resourceStackRead()` function, which sets the ID to an empty string if no stack is found. To solve this I've pulled out query for a stack and updating the state based on a stack struct into separate functions, and I've updated the import handling to detect when the stack can't be found an output a decent error message.

I pulled the `PopulateStack()` function into the structs package to follow the pattern we've taken for the scheduled task structs.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

## Related issues

- Fixes [#405](https://github.com/spacelift-io/terraform-provider-spacelift/issues/405)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development
- [x] Examples for new resources and data sources have been added
- [x] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [ ] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [x] Reviewers have been assigned
- [x] Changes have been reviewed by at least one other engineer
- [x] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

**Comment by marcinwyszynski** [comment]
Task linked: [CU-862j01umm Importing stack fails if stack ID does not exist](https://app.clickup.com/t/862j01umm)

File: spacelift/resource_stack.go [hunk 1]
```diff
@@ -914,13 +847,22 @@ func ignoreOnceCreated(_, _, _ string, d *schema.ResourceData) bool {
...
 
-	diag := resourceStackRead(ctx, d, meta)
-	if diag.HasError() {
-		return nil, fmt.Errorf("could not import stack: %v", diag)
+	stack, err := getStackByID(ctx, meta.(*internal.Client), stackID)
+	if err != nil {
+		return nil, fmt.Errorf("Could not query for stack with ID %q: %v", stackID, err)
+	}
+
+	if stack == nil {
```
Comment (marcinwyszynski) [inline spacelift/resource_stack.go:860]: Would it make sense to also suggest that the stack may exist but you have no access to it?

Comment (adamconnelly) [reply spacelift/resource_stack.go:860]: Sure - I'll update the message.

Comment (adamconnelly) [reply spacelift/resource_stack.go:860]: Ok - that's the message updated.

**Pull request merged**